app/models/issue.rb

```diff
@@ -45,4 +45,16 @@ def lt_ln_present?(obj)
   def status_name
     STATUSES[status]
   end
+
+  def published?
+    (status == 'open') || (status == 'closed')
+  end
+
+  def can_read_by?(user)
+    if user.present?
+      user.admin? || user.moderator? || (self.user == user) || published?
+    else
+      false
+    end
+  end
```

Comment:
Kostya but in this case it won't be possible to do something like
```
if current_user
   ...
```

and existing code needs to be changed. I see no benefits of adding new class Guest. What it will do?